### PR TITLE
INFRA-333 Support RESEARCH2 context

### PR DESCRIPTION
**Pull request by mdenburger**
Handles pipeline context `RESEARCH2` like `RESEARCH`.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -32,7 +32,7 @@
         <main.class>com.hartwig.snpcheck.SnpCheckMain</main.class>
         <jackson-core.version>2.14.2</jackson-core.version>
         <hmf-api-java-client.version>3.2.4</hmf-api-java-client.version>
-        <app.hartwig-events.version>3.14.0</app.hartwig-events.version>
+        <app.hartwig-events.version>3.15.0-alpha.2</app.hartwig-events.version>
```
Comment (mvanniekerkHartwig) [inline pom.xml:35]: Upgrade to release version before merging.

File: src/main/java/com/hartwig/snpcheck/SnpCheck.java [hunk 2]
```diff
@@ -81,14 +85,18 @@ public void handle(final PipelineComplete event) {
...
-            if (event.pipeline().context().equals(Pipeline.Context.RESEARCH)) {
+            if (isResearchContext(event.pipeline().context())) {
                 validateResearchWhenSourceRunValidated(event, run);
             } else {
                 validateRunWithSnpcheck(event, run);
             }
         }
     }
 
+    private boolean isResearchContext(Pipeline.Context context) {
```
Comment (mvanniekerkHartwig) [inline src/main/java/com/hartwig/snpcheck/SnpCheck.java:96]: can be static.

File: src/test/java/com/hartwig/snpcheck/SnpCheckTest.java [hunk 3]
```diff
@@ -1,57 +1,40 @@
 package com.hartwig.snpcheck;
 
-import static java.util.Collections.emptyList;
```
Comment (mvanniekerkHartwig) [inline src/test/java/com/hartwig/snpcheck/SnpCheckTest.java:3]: Wrong code style

**Review by mvanniekerkHartwig — APPROVED** [review]
Few nitpicks, but I'll approve

File: src/test/java/com/hartwig/snpcheck/SnpCheckTest.java (hunk 3 above)
Comment (mdenburger) [reply src/test/java/com/hartwig/snpcheck/SnpCheckTest.java]: 👍  fixed. No idea why these change sometimes 🤔

**Pull request merged**